src/xcode/ENA/ENA/Source/Scenes/Home/Cells/Statistics/Manage Statistics/ManageStatisticsCardView.swift

```diff
@@ -0,0 +1,167 @@
+		stackView.arrangedSubviews.forEach { subview in
+			stackView.removeArrangedSubview(subview)
+			subview.removeFromSuperview()
+		}
+
+		let addView = { () -> CustomDashedView in
+			let add = CustomDashedView.instance(for: .add)
+			add.tapHandler = { [weak self] in
+				Log.debug("add…", log: .ui)
+				self?.onAddLocalIncidenceButtonPressed(add)
+			}
+			add.label.onAccessibilityFocus = onAccessibilityFocus
+			return add
+		}()
+
+		let modifyView = { () -> CustomDashedView in
+			let modify = CustomDashedView.instance(for: .modify)
+			modify.tapHandler = { [weak self] in
+				Log.debug("modify…", log: .ui)
+				self?.onEditButtonTap?()
+			}
+			modify.label.onAccessibilityFocus = onAccessibilityFocus
+			return modify
+		}()
+
+		switch state {
+		case .empty:
+			// just 'add'
+			stackView.addArrangedSubview(addView)
+		case .notYetFull:
+			// 'add' & 'modify'
+			stackView.addArrangedSubview(addView)
+			stackView.addArrangedSubview(modifyView)
+		case .full:
+			// just 'modify'
+			stackView.addArrangedSubview(modifyView)
+		}
+	}
+	
+	// MARK: - Private
+
+	private func presentAddLocalStatistics(selectValueViewModel: SelectValueViewModel) {
+		let selectValueViewController = SelectValueTableViewController(
+			selectValueViewModel,
+			closeOnSelection: false,
+			dismiss: { [weak self] in
+				self?.onDismissState?()
+			})
+		onAddStateButtonTap?(selectValueViewController)
+	}
+
+	private func presentAddLocalStatisticsDistrict(selectValueViewModel: SelectValueViewModel) {
+		let selectValueViewController = SelectValueTableViewController(
+			selectValueViewModel,
+			closeOnSelection: true,
+			dismiss: { [weak self] in
+				let dismissToRoot = self?.viewModel?.district != nil
+				self?.onDismissDistrict?(dismissToRoot)
+			}
+		)
+		onAddDistrict?(selectValueViewController)
+	}
+
+	private var onAddStateButtonTap: ((SelectValueTableViewController) -> Void)?
+	private var onAddDistrict: ((SelectValueTableViewController) -> Void)?
+	private var onDismissState: (() -> Void)?
+	private var onDismissDistrict: ((Bool) -> Void)?
+	private var onFetchGroupData: ((LocalStatisticsDistrict) -> Void)?
+	private var onEditButtonTap: (() -> Void)?
+	private var onAccessibilityFocus: (() -> Void)?
+	private var viewModel: ManageStatisticsCardsViewModel?
+	private var model: LocalStatisticsModel?
+}
+
+enum LocalStatisticsState {
+	/// No local stats selected
+	case empty
+	/// Can add more local statistics
+	case notYetFull
+	/// The maximum number of local statistics selected
+	case full
+
+	static func with(_ store: LocalStatisticsCaching) -> Self {
+		switch store.selectedLocalStatisticsDistricts.count {
+		case ...0:
+			return .empty
+		case 1...(Self.threshold - 1):
+			return .notYetFull
+		case Self.threshold...: // allow over threshold values in favor to fatal errors
+			return .full
+		default:
+			fatalError("should not happen™")
+		}
+	}
+
+	/// Maximum number of custom statistics
+	private static let threshold = 5
+}
```

Comment:
Should be in a separate file if used in multiple places or be put inside the view model if only needed inside of it